test/tests_for_sdkmr/EE2Runjob_test.py

```diff
@@ -0,0 +1,268 @@
+    slack = create_autospec(SlackClient, spec_set=True, instance=True)
+    sdkmr.get_logger.return_value = logger
+    sdkmr.get_workspace.return_value = ws
+    sdkmr.get_catalog_utils.return_value = catutils
+    catutils.get_catalog.return_value = catalog
+    sdkmr.get_condor.return_value = condor
+    sdkmr.get_kafka_client.return_value = kafka
+    sdkmr.get_slack_client.return_value = slack
+    sdkmr.get_mongo_util.return_value = mongo
+    sdkmr.get_user_id.return_value = "someuser"
+    sdkmr.get_token.return_value = "tokentokentoken"
+    sdkmr.check_as_admin.return_value = True
+    sdkmr.save_job.return_value = "603051cfaf2e3401b0500982"
+    ws.get_object_info3.return_value = {"paths": [["1/2/3"], ["4/5/6"]]}
+    catalog_resources = {
+        "client_group": "grotesquememlong",
+        "request_cpus": "4",
+        "request_memory": "32",
+    }
+    catutils.get_normalized_resources.return_value = catalog_resources
+    condor.extract_resources.return_value = CondorResources(
+        "4", "2600", "32", "grotesquememlong"
+    )
+    catalog.get_module_version.return_value = {"git_commit_hash": "git5678"}
+    condor.run_job.return_value = SubmissionInfo("cluster42", {}, None)
+    retjob = Job()
+    retjob.id = ObjectId("603051cfaf2e3401b0500982")
+    retjob.status = "created"
+    mongo.get_job.return_value = retjob
+
+    # set up the class to be tested and run the method
+    rj = EE2RunJob(sdkmr)
+    params = {
+        "method": "lolcats.lol_unto_death",
+        "source_ws_objects": ["1/2/3", "4/5/6"],
+    }
+    assert rj.run(params, as_admin=True) == "603051cfaf2e3401b0500982"
+
+    # check mocks called as expected
+    sdkmr.check_as_admin.assert_called_once_with(JobPermissions.WRITE)
+    ws.get_object_info3.assert_called_once_with(
+        {"objects": [{"ref": "1/2/3"}, {"ref": "4/5/6"}], "ignoreErrors": 1}
+    )
+    catutils.get_normalized_resources.assert_called_once_with("lolcats.lol_unto_death")
+    condor.extract_resources.assert_called_once_with(catalog_resources)
+    catalog.get_module_version.assert_called_once_with(
+        {"module_name": "lolcats", "version": "release"}
+    )
+
+    # initial job data save
+    expected_job = Job()
+    expected_job.user = "someuser"
+    expected_job.status = "created"
+    ji = JobInput()
+    ji.method = "lolcats.lol_unto_death"
+    ji.service_ver = "git5678"
+    ji.source_ws_objects = ["1/2/3", "4/5/6"]
+    ji.parent_job_id = "None"
+    jr = JobRequirements()
+    jr.clientgroup = "grotesquememlong"
+    jr.cpu = "4"
+    jr.memory = "3"
+    jr.disk = "26"
+    ji.requirements = jr
+    ji.narrative_cell_info = Meta()
+    expected_job.job_input = ji
+    assert len(sdkmr.save_job.call_args_list) == 2
+    got_job = sdkmr.save_job.call_args_list[0][0][0]
+    assert_jobs_equal(got_job, expected_job)
+
+    kafka.send_kafka_message.assert_any_call(
+        KafkaCreateJob("someuser", "603051cfaf2e3401b0500982")
```

Comment:
can you make `"603051cfaf2e3401b0500982"` into a variable since it's used numerous times?

Reply:
Actually, I'd do that for all the non-trivial values that you have use in the Job / JobInput / JobRequirements objects and the condor params object below.

Reply:
done

Reply:
Huh, that's weird, there was only one comment when I wrote `done`. It's not done yet.

Reply:
done